packages/core/src/shared/utilities/textUtilities.ts

```diff
@@ -395,3 +405,34 @@ export function undefinedIfEmpty(str: string | undefined): string | undefined {
 
     return undefined
 }
+
+/**
+ * Extracts the file path and selection context from the message.
+ *
+ * @param {any} message - The message object containing the file and selection context.
+ * @returns {Object} - An object with `filePath` and `selection` properties.
+ */
+export function extractFileAndCodeSelectionFromMessage(message: any) {
+    const filePath = message?.context?.activeFileContext?.filePath
+    const selection = message?.context?.focusAreaContext?.selectionInsideExtendedCodeBlock as vscode.Selection
+    return { filePath, selection }
+}
+
+/**
+ * Indents the given code based on the current document's indentation at the selection start.
+ *
+ * @param {any} message - The message object containing the code.
+ * @param {vscode.TextDocument} doc - The VSCode document where the code is applied.
+ * @param {vscode.Selection} selection - The selection range in the document.
+ * @returns {string} - The processed code to be applied to the document.
+ */
+export function getIndentedCode(message: any, doc: vscode.TextDocument, selection: vscode.Selection) {
```

Comment:
should this live in `packages/core/src/shared/utilities/textDocumentUtilities.ts`